From 76ede226f95cf6aa3f72db1ca40004e4b3d59dd3 Mon Sep 17 00:00:00 2001
From: Peng Fan <peng.fan@nxp.com>
Date: Mon, 23 Jul 2018 14:44:37 +0800
Subject: [PATCH 4220/5242] MLK-19003-2 ARM64: dts: imx8mq: inmate: add pci
 host ecam generic node

commit  3eb2742156ba9889c13139d0f56b5a51efbbfa12 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add pci node to let inmate could detect ivshmem net.
The reg base is the PCI_CFG_BASE in jailhouse hypervisor.
The interrupt needs to be same vpci_irq_base + [x] in jailhouse
hypevisor.

In this patch also delete memory node, jailhosue cell linux
tool also creates a memory node, there are two memory nodes,
cause issues when using ivshmem-net.

Signed-off-by: Peng Fan <peng.fan@nxp.com>
Cc: Richard Zhu <hongxing.zhu@nxp.com>
Reviewed-by: Bai Ping <ping.bai@nxp.com>
(cherry picked from commit e1f59c589d16a556310e5a12d62e4a1913fc905a)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8mq-evk-inmate.dts   |   15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-inmate.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-inmate.dts
index 8451ee5..0914012 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-inmate.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-inmate.dts
@@ -66,8 +66,23 @@
 	display-subsystem {
 		/delete-property/ compatible;
 	};
+
+	pci@bfb00000 {
+		compatible = "pci-host-ecam-generic";
+		device_type = "pci";
+		bus-range = <0 0>;
+		#address-cells = <3>;
+		#size-cells = <2>;
+		#interrupt-cells = <1>;
+		interrupt-map-mask = <0 0 0 7>;
+		interrupt-map = <0 0 0 1 &gic GIC_SPI 109 IRQ_TYPE_EDGE_RISING>;
+		reg = <0x0 0xbfb00000 0x0 0x100000>;
+		ranges = <0x02000000 0x00 0x10000000 0x0 0x10000000 0x00 0x10000>;
+	};
 };
 
+/delete-node/ &{/memory@40000000};
+
 &clk {
 	/delete-property/ compatible;
 };
-- 
1.7.9.5

